filebeat/inputsource/tcp/client.go

```diff
@@ -0,0 +1,79 @@
+package tcp
+
+import (
+	"bufio"
+	"net"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+// Client is a remote client.
+type client struct {
+	conn           net.Conn
+	log            *logp.Logger
+	callback       CallbackFunc
+	done           chan struct{}
+	metadata       Metadata
+	splitFunc      bufio.SplitFunc
+	maxReadMessage size
+	timeout        time.Duration
+}
+
+func newClient(
+	conn net.Conn,
+	log *logp.Logger,
+	callback CallbackFunc,
+	splitFunc bufio.SplitFunc,
+	maxReadMessage size,
+	timeout time.Duration,
+) *client {
+	client := &client{
+		conn:           conn,
+		log:            log.With("address", conn.RemoteAddr()),
+		callback:       callback,
+		done:           make(chan struct{}),
+		splitFunc:      splitFunc,
+		maxReadMessage: maxReadMessage,
+		timeout:        timeout,
+		metadata: Metadata{
+			IPAddress: conn.RemoteAddr().String(),
```

Comment:
The way this is populated it's more than an IP address so I think the name is confusing. I'd go with `RemoteAddr`. Or [split](https://golang.org/pkg/net/#SplitHostPort) it here and have two fields `RemoteIP` and `RemotePort`.

Reply:
If I understand this correctly, the ip address provided here is in the end the host that sent the log message. So my suggestion would be to use `host.ip` for this. So the `host.ip` would be the same if the file is read locally or sent over tcp.

For the `port` I'm not sure if this is relevant in this context and could be skipped. If we add the port, it would be the port of the service running, so `service.port`?

Reply:
Maybe you are right port in that context isn't necessary and we can drop it and I would also say the same for port of the service running. 

@ruflin For the future I still believe that `source.ip` is what we should aim, I think in using `source` as the IP until ECS is deployed is a good compromised into not creating new fields that we have to deprecated right away.

Reply:
agreement on zoom: 
- set remote ip + remote port to the `source` field in the event.
- on Metadata uses `RemoteAddr` instead of `IPAdress`.

Example format : `127.0.0.1:99998`